### PR TITLE
Add a scientist NPC to the refugee center and a mission to find central lab

### PR DESCRIPTION
**Pull request by BrianLefler**
#### Summary

SUMMARY: Features "Add lab mission chain."

#### Purpose of change
Start of a mission chain to add lore around the labs and give missions involving them.

#### Describe the solution
There is an NPC at the refugee center with the Doctor class.

There are now 5 missions.  The three middle ones are a repetitive and could absolutely be made more interesting, but they at least take a player through the available lab content.

- Run a blood sample to the hospital.
- Get data off the closest lab.
- Get data from a random hidden basement lab.
- Get data from the 4th level of an ice lab.
- Get maps from a lab train depot.

The get data quests will activate/create 3 consoles randomly on a given lab level for the player to find.

There's some common functionality created for this: Sticking lab computers down, finding the entrance that corresponds to an underground lab spot, mapping the route between player and lab.

This also adds some more broken consoles to lab maps - when a quest wants to create a lab computer for the player to interact with, we'll prefer to activate pre-existing broken consoles rather than drop new ones down (but will if necessary).  Better code would check for multiple maps to see if they have consoles or maybe rewrite the entire content of the map via target_om_ter_random_or_create, but the current logic is functional enough.

This also fixes an off by 1 error in find_closest.  The code builds increasingly large rectangles and scans across them, but would fail to check the last tile in a side of the rectangle, creating 4 line-shaped blind spots where we'd fail to check terrain.  I noticed it when the nearest lab was occasionally an unexpectedly far away one (because the expected lab was in the blind spot).

### CODE REVIEW

**Comment by FulcrumA** [comment]
I'd really hope for such scientist to be introduced as something that shows up over time, rather than immediately "oh hey, you survived, we need your help" thing which seems a bit tacked on.
I like more NPCs and quests, but I'd prefer the interaction to happen more naturally than to make it blatantly that "this NPC's reason for existence is to give you a quest".

File: data/json/npcs/missiondef.json [hunk 1]
```diff
@@ -834,6 +834,28 @@
...
+    "value": 50000,
+    "item": "software_train_data",
+    "start": "reveal_lab_train_depot",
+    "origins": [ "ORIGIN_SECONDARY" ],
+    "dialogue": {
+      "describe": "We need help...",
+      "offer": "I met a construction worker who says there was a big contract for tunneling equipment eight months ago.  He gave me an address to a government lab where they delivered the equipment.  I want you to go there, find the tunnels, and download their layout.  Maybe we can use it for safer transport.",
+      "accepted": "So glad for your help.",
+      "rejected": "Can't blame you, but come back if you change your mind.",
+      "advice": "Come prepared: That tunnel entrance is going to be deeper inside a lab than a normal subway.  Maybe four stories down.",
```
Comment (Night-Pryanik) [inline data/json/npcs/missiondef.json]: Decapitalize "That".

File: src/mission_start.cpp [hunk 2]
```diff
@@ -1780,3 +1780,41 @@ void mission_start::reveal_refugee_center( mission *miss )
...
+    }
+    g->u.i_add( item( "usb_drive", 0 ) );
+    add_msg( _( "%s gave you a USB drive." ), dev->name.c_str() );
+
+    tinymap compmap;
+    compmap.load( place.x * 2, place.y * 2, place.z, false );
+    tripoint comppoint;
+
+    bool comp_found = false;
+    for( int x = 0; x < SEEX * 2 && !comp_found; x++ ) {
```
Comment (Night-Pryanik) [inline src/mission_start.cpp]: Maybe replace this check with `g->m.points_in_radius`?

**Comment by BrianLefler** [comment]
Ok, I now have a 4 mission chain.  These missions are not particularly well conceived, but I think they're better than nothing.  If someone wants to merge it now they can, but I'm also going to continue to iterate on this.

1) Gallon of disinfectant (to free up doctors to have time to do research).
2) Sample of zombie blood, analyze at a hospital.
3) Download lab archives.
4) Find lab tunnels.

**Comment by BrianLefler** [comment]
Ok, I think this is at a good stopping point.

There are now 5 missions.  They are a little repetitive and could absolutely be made more interesting, but they at least take a player through the available lab content.

- Run a blood sample to the hospital.
- Get data off the closest lab.
- Get data from a random hidden basement lab.
- Get data from the 4th level of an ice lab.
- Get maps from a lab train depot.

The get data quests will active/create 3 consoles randomly on a given lab level for the player to find.

There's some common functionality created for this: Sticking lab computers down, finding the entrance that corresponds to an underground lab spot.

This also adds some more broken consoles to lab maps - when a quest wants to create a lab computer for the player to interact with, we'll prefer to activate pre-existing broken consoles rather than drop new ones down (but will if necessary).  Better code would check for multiple maps to see if they have consoles or maybe rewrite the entire content of the map via target_om_ter_random_or_create, but the current logic is functional enough.

This also fixes an off by 1 error in find_closest.  The code builds increasingly large rectangles and scans across them, but would fail to check the last tile in a side of the rectangle, creating 4 line-shaped blind spots where we'd fail to check terrain.  I noticed it when the nearest lab was occasionally an unexpectedly far away one (because the expected lab was in the blind spot).

**Comment by FulcrumA** [comment]
Jenkins failed at summary. I believe it's the lack of quotation marks.

**Comment by BrianLefler** [comment]
Ok, I don't know what the build wants.

20:24:52 Invalid Summary line, please provide a summary similar to the following:
20:24:52 SUMMARY: Features "Added ability to fly."

Actual line in summary desc is:
SUMMARY: Features "Add a NPC to the refugee center and lab mission chain."

**Comment by ZhilkinSerg** [comment]
There were some issues with summary line parsing.

Jenkins rebuild.

**Comment by AMurkin** [comment]
No text should go after None in summary. Or change it to another type: Content - for example.

**Comment by ZhilkinSerg** [comment]
"Kill Raider Mission" overmap terrain spawned trimmed.

image

**Pull request merged**